packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -333,3 +333,22 @@ function srcset_url_equal(element, srcset) {
 		)
 	);
 }
+
+/**
+ * @param {HTMLImageElement} element
+ * @returns {void}
+ */
+export function handle_ff_lazy_img(element) {
+	// If we're using Firefox and the image has a lazy loading attribute, we need to
+	// apply this attribute, along with the src, after it has been appended to the document
+	// otherwise the lazy behaviour will not work due to our cloneNode heuristic for templates.
+	if (/Firefox/.test(navigator.userAgent) && element.getAttribute('loading') === 'lazy') {
+		var src = element.src;
+		element.removeAttribute('loading');
+		element.removeAttribute('src');
+		requestIdleCallback(() => {
```

Comment:
is `requestIdleCallback` the right thing to use here? could it result in a user-noticeable period with no `src` (which could affect e.g. CSS selectors) or is it guaranteed not to?

Reply:
I switched to `requestAnimationFrame`, I wonder if that would be fine here?